elyra/cli/pipeline_app.py

```diff
@@ -526,34 +521,64 @@ def describe(json_option, pipeline_path):
+    describe_dict[pipeline_constants.RUNTIME_IMAGE] = set()
     for node in primary_pipeline.nodes:
         # collect information about file dependencies
         for dependency in node.get_component_parameter("dependencies", []):
             describe_dict["file_dependencies"].add(f"{dependency}")
         # collect information about component dependencies
         if node.component_source is not None:
             describe_dict["component_dependencies"].add(node.component_source)
+        # collect information of mounted volumes
+        for mounted_volume in node.get_component_parameter(pipeline_constants.MOUNTED_VOLUMES, []):
+            temp_mount_value = mounted_volume.split("=")[-1]
+            if temp_mount_value != "":
+                describe_dict[pipeline_constants.MOUNTED_VOLUMES].add(f"{temp_mount_value}")
+        # collection runtime image details
+        temp_runtime_image_value = node.get_component_parameter(pipeline_constants.RUNTIME_IMAGE)
+        if temp_runtime_image_value:
+            describe_dict[pipeline_constants.RUNTIME_IMAGE].add(f"{temp_runtime_image_value}")
+
+        # collect notebook / script name when pipeline is generic
+        if describe_dict["type"] is None:
+            temp_value = node.get_component_parameter("filename")
+            if not temp_value:
+                pass
+            elif Path(Path(temp_value).name).suffix == ".ipynb":
+                describe_dict["notebooks"].add(temp_value)
+            else:
+                describe_dict["scripts"].add(temp_value)
 
     if not json_option:
-        for key in pipeline_keys:
+        click.echo()
+
+        print_banner("Elyra Pipeline details")
+        for key in list(describe_dict):
             readable_key = " ".join(key.title().split("_"))
-            if key in iter_keys:
+            if isinstance(describe_dict[key], set):
                 click.echo(f"{readable_key}:")
-                if describe_dict.get(key, set()) == set():
+                # if describe_dict.get(key, set()) == set():
+                if not describe_dict.get(key):
                     click.echo(f"{' ' * indent_length}{blank_list[0]}")
                 else:
                     for item in describe_dict.get(key, blank_list):
                         click.echo(f"{' ' * indent_length}- {item}")
             else:
                 click.echo(f"{readable_key}: {describe_dict.get(key, blank_field)}")
     else:
-        for key in iter_keys:
-            describe_dict[key] = list(describe_dict[key])
-        for key in pipeline_keys:
+        # for key in list(describe_dict.keys()):
+        for key in list(describe_dict):
+            if isinstance(describe_dict[key], set):
+                describe_dict[key] = list(describe_dict[key])
+
             value = describe_dict.get(key)
-            if value is None or (key in iter_keys and len(value) == 0):
+            # if value is None or (key in list(describe_dict.keys()) and len(str(value)) == 0):
```

Comment:
I'm sure you're already tracking this, but just in case, this line can probably be removed now, as well as 574 and 565 🙂

Reply:
Resolved